### PR TITLE
Refactor JMS/Serializer annotations to attributes

### PR DESCRIPTION
**Pull request by TheCadien**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no 
| Related tickets | #7156 
| License | MIT

### CODE REVIEW

**Comment by alexander-schranz** [comment]
@TheCadien looks like the `lowest` version is erroring. Think we need increase the min requirement for jms serializer or/and jms serializer bundle to the first version supporting attributes.

**Comment by TheCadien** [comment]
@alexander-schranz the support for attributes in JMS Serializer comes with 1.14. 
The JMS Bundle min requirement for that begins with 4.0. 
So i drop the 3.8

File: src/Sulu/Component/Rest/ListBuilder/CollectionRepresentation.php [hunk 1]
```diff
@@ -13,11 +13,7 @@
 
 use JMS\Serializer\Annotation as Serializer;
 
-/**
- * @Serializer\ExclusionPolicy("all")
- *
- * This class represents a list for our common rest services.
- */
+#[Serializer\ExclusionPolicy('all')] // This class represents a list for our common rest services.
```
Comment (alexander-schranz) [inline src/Sulu/Component/Rest/ListBuilder/CollectionRepresentation.php]:
```suggestion
/**
 * This class represents a list for our common rest services.
 */
#[Serializer\ExclusionPolicy('all')]
```

File: src/Sulu/Component/Rest/ListBuilder/PaginatedRepresentation.php [hunk 2]
```diff
@@ -13,11 +13,7 @@
 
 use JMS\Serializer\Annotation as Serializer;
 
-/**
- * @Serializer\ExclusionPolicy("all")
- *
- * This class represents a list for our common rest services.
- */
+#[Serializer\ExclusionPolicy('all')] // This class represents a list for our common rest services.
```
Comment (alexander-schranz) [inline src/Sulu/Component/Rest/ListBuilder/PaginatedRepresentation.php]:
```suggestion
/**
 * This class represents a list for our common rest services.
 */
#[Serializer\ExclusionPolicy('all')]
```

File: src/Sulu/Component/SmartContent/Rest/ItemCollectionRepresentation.php [hunk 3]
```diff
@@ -15,11 +15,7 @@
 use Sulu\Component\Rest\ListBuilder\CollectionRepresentation;
 use Sulu\Component\SmartContent\DatasourceItemInterface;
 
-/**
- * @Serializer\ExclusionPolicy("all")
- *
- * Provides a collection of items and the datasource.
- */
+#[Serializer\ExclusionPolicy('all')] // Provides a collection of items and the datasource.
```
Comment (alexander-schranz) [inline src/Sulu/Component/SmartContent/Rest/ItemCollectionRepresentation.php]:
```suggestion
/**
 * Provides a collection of items and the datasource.
 */
#[Serializer\ExclusionPolicy('all')]
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Comment by alexander-schranz** [comment]
@TheCadien always the problems with finding out which prefer-lowest version is really compatible if dependency defines there [optional dependencies](https://github.com/alexander-schranz/composer-optional-dependencies) not correctly.

In this case we need increase fos rest also to `^3.1.0` because of: https://github.com/FriendsOfSymfony/FOSRestBundle/commit/f2904a98ee8a4b383f65f1e8c348e28da8c0ad3f

**Comment by TheCadien** [comment]
@alexander-schranz  ahhh i see ! i take a look !

**Comment by alexander-schranz** [comment]
@TheCadien This case I think it only should be the fos rest requirement think there is no other jms related dependency:

```diff
-        "friendsofsymfony/rest-bundle": "^3.0.4",
+        "friendsofsymfony/rest-bundle": "^3.1",
```

File: composer.json [hunk 4]
```diff
@@ -47,14 +47,14 @@
...
-        "friendsofsymfony/rest-bundle": "^3.0.4",
+        "friendsofsymfony/rest-bundle": "^3.1",
         "gedmo/doctrine-extensions": "^3.0.4",
         "guzzlehttp/promises": "^1.0",
         "handcraftedinthealps/goodby-csv": "^1.4",
         "handcraftedinthealps/rest-routing-bundle": "^1.0",
         "imagine/imagine": "^1.0",
         "jms/serializer": "^3.16",
-        "jms/serializer-bundle": "^4.0 || ^5.0",
+        "jms/serializer-bundle": "^3.8 || ^4.0 || ^5.0",
```
Comment (alexander-schranz) [inline composer.json]: think you accidently reverted this one

```suggestion
        "jms/serializer-bundle": "^4.0 || ^5.0",
```

**Comment by TheCadien** [comment]
@alexander-schranz looks like twig has also an depedency :D i check that

**Comment by alexander-schranz** [comment]
@TheCadien seems like a bug in serializer bundle `4.0.0` should be fixed in `^4.0.1`: https://github.com/schmittjoh/JMSSerializerBundle/compare/4.0.0...4.0.1

File: composer.json [hunk 5]
```diff
@@ -47,14 +47,14 @@
...
-        "friendsofsymfony/rest-bundle": "^3.0.4",
+        "friendsofsymfony/rest-bundle": "^3.1",
         "gedmo/doctrine-extensions": "^3.0.4",
         "guzzlehttp/promises": "^1.0",
         "handcraftedinthealps/goodby-csv": "^1.4",
         "handcraftedinthealps/rest-routing-bundle": "^1.0",
         "imagine/imagine": "^1.0",
         "jms/serializer": "^3.16",
-        "jms/serializer-bundle": "^3.8 || ^4.0 || ^5.0",
+        "jms/serializer-bundle": "^4.0 || ^5.0",
```
Comment (alexander-schranz) [inline composer.json]:
```suggestion
        "jms/serializer-bundle": "^4.0.1 || ^5.0",
```

**Comment by alexander-schranz** [comment]
Finding out which `lowest` version plays well with the other dependencies is always harder as upgrading dependency to a new major 😄

**Comment by TheCadien** [comment]
@alexander-schranz hate that :D

File: UPGRADE.md [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # Upgrade
 
 ## 2.6.0
-
+
```
Comment (alexander-schranz) [inline UPGRADE.md]:
```suggestion

```

**Pull request merged**